### DISCUSSION TITLE
QuickSort reinvented - beating most implementations in pure C by a significant margin

### DISCUSSION THREAD

**Discussion by dumblob** [vlang/v#11675] [Ideas]
For those hungry after speed, I'd recommend recent observation why finally "correctly" implement QuickSort (in pure C - without any assembly - with very short code base - about 3x as minimal quicksort - but leveraging modern CPU pipelines), despite being unstable, still matters and why MergeSort/HeapSort (and their variants - e.g. [quadsort](https://github.com/scandum/quadsort )) can't beat it for primitive values/types:

https://github.com/gerben-s/quicksort-blog-post/blob/master/blog.md

(note the linked quadsort above seems still slower than the mergesort implemented in the above-linked blog post)

**Comment by dumblob** [comment]
@penguindark @ka-weihe just pinging as I've noticed you were interested in optimizations (regex, hash map, ...)

**Comment by penguindark** [comment]
Very interesting! Tnx :)
